From 995b33c745085439d3cb3fed85dbfdb89cb876a9 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenko@src.gnome.org>
Date: Sat, 8 Jul 2017 19:37:51 +0200
Subject: remove
 ggit_submodule_update_options_set/get_clone_checkout_strategy()

GIT_CHECKOUT_SAFE_CREATE was folded into GIT_CHECKOUT_SAFE in checkout_opts.

Signed-off-by: Igor Gnatenko <ignatenko@src.gnome.org>

https://bugzilla.gnome.org/show_bug.cgi?id=784706
---
 libgit2-glib/ggit-submodule-update-options.c | 65 ----------------------------
 libgit2-glib/ggit-submodule-update-options.h |  4 --
 2 files changed, 69 deletions(-)

diff --git a/libgit2-glib/ggit-submodule-update-options.c b/libgit2-glib/ggit-submodule-update-options.c
index 3b092b1..94ef550 100644
--- libgit2-glib/ggit-submodule-update-options.c
+++ libgit2-glib/ggit-submodule-update-options.c
@@ -44,7 +44,6 @@ enum
 	PROP_0,
 	PROP_CHECKOUT_OPTIONS,
 	PROP_FETCH_OPTIONS,
-	PROP_CLONE_CHECKOUT_STRATEGY,
 };
 
 static void
@@ -79,10 +78,6 @@ ggit_submodule_update_options_set_property (GObject      *object,
 		ggit_submodule_update_options_set_fetch_options (options,
 		                                                 g_value_get_boxed (value));
 		break;
-	case PROP_CLONE_CHECKOUT_STRATEGY:
-		ggit_submodule_update_options_set_clone_checkout_strategy (options,
-		                                                           g_value_get_flags (value));
-		break;
 	default:
 		G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
 		break;
@@ -108,10 +103,6 @@ ggit_submodule_update_options_get_property (GObject    *object,
 	case PROP_FETCH_OPTIONS:
 		g_value_set_boxed (value, priv->fetch_options);
 		break;
-	case PROP_CLONE_CHECKOUT_STRATEGY:
-		g_value_set_flags (value,
-		                   ggit_submodule_update_options_get_clone_checkout_strategy (options));
-		break;
 	default:
 		G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
 		break;
@@ -144,16 +135,6 @@ ggit_submodule_update_options_class_init (GgitSubmoduleUpdateOptionsClass *klass
 	                                                     GGIT_TYPE_FETCH_OPTIONS,
 	                                                     G_PARAM_READWRITE |
 	                                                     G_PARAM_STATIC_STRINGS));
-
-	g_object_class_install_property (object_class,
-	                                 PROP_CLONE_CHECKOUT_STRATEGY,
-	                                 g_param_spec_flags ("clone-checkout-strategy",
-	                                                     "Clone Checkout Strategy",
-	                                                     "Clone checkout strategy",
-	                                                     GGIT_TYPE_CHECKOUT_STRATEGY,
-	                                                     GGIT_CHECKOUT_NONE,
-	                                                     G_PARAM_READWRITE |
-	                                                     G_PARAM_STATIC_STRINGS));
 }
 
 static void
@@ -318,50 +299,4 @@ ggit_submodule_update_options_set_fetch_options (GgitSubmoduleUpdateOptions *opt
 	g_object_notify (G_OBJECT (options), "fetch-options");
 }
 
-/**
- * ggit_submodule_update_options_get_clone_checkout_strategy:
- * @options: a #GgitSubmoduleUpdateOptions.
- *
- * Gets the clone checkout strategy.
- *
- * Returns: the clone checkout strategy.
- */
-GgitCheckoutStrategy
-ggit_submodule_update_options_get_clone_checkout_strategy (GgitSubmoduleUpdateOptions *options)
-{
-	GgitSubmoduleUpdateOptionsPrivate *priv;
-
-	g_return_val_if_fail (GGIT_IS_SUBMODULE_UPDATE_OPTIONS (options), GGIT_CHECKOUT_NONE);
-
-	priv = ggit_submodule_update_options_get_instance_private (options);
-
-	return priv->options.clone_checkout_strategy;
-}
-
-/**
- * ggit_submodule_update_options_set_clone_checkout_strategy:
- * @options: a #GgitSubmoduleUpdateOptions.
- * @checkout_strategy: a #GgitCheckoutStrategy.
- *
- * Sets the clone checkout strategy. Use %GGIT_CHECKOUT_SAFE_CREATE to create
- * all files in the working directory for the newly cloned repository.
- */
-void
-ggit_submodule_update_options_set_clone_checkout_strategy (GgitSubmoduleUpdateOptions *options,
-                                                           GgitCheckoutStrategy        checkout_strategy)
-{
-	GgitSubmoduleUpdateOptionsPrivate *priv;
-
-	g_return_if_fail (GGIT_IS_SUBMODULE_UPDATE_OPTIONS (options));
-
-	priv = ggit_submodule_update_options_get_instance_private (options);
-
-	if (priv->options.clone_checkout_strategy != checkout_strategy)
-	{
-		priv->options.clone_checkout_strategy = checkout_strategy;
-
-		g_object_notify (G_OBJECT (options), "clone-checkout-strategy");
-	}
-}
-
 /* ex:set ts=8 noet: */
diff --git a/libgit2-glib/ggit-submodule-update-options.h b/libgit2-glib/ggit-submodule-update-options.h
index 5816d97..d26290b 100644
--- libgit2-glib/ggit-submodule-update-options.h
+++ libgit2-glib/ggit-submodule-update-options.h
@@ -50,10 +50,6 @@ GgitFetchOptions      *ggit_submodule_update_options_get_fetch_options     (Ggit
 void                   ggit_submodule_update_options_set_fetch_options     (GgitSubmoduleUpdateOptions *options,
                                                                             GgitFetchOptions           *fetch_options);
 
-GgitCheckoutStrategy   ggit_submodule_update_options_get_clone_checkout_strategy (GgitSubmoduleUpdateOptions *options);
-void                   ggit_submodule_update_options_set_clone_checkout_strategy (GgitSubmoduleUpdateOptions *options,
-                                                                                  GgitCheckoutStrategy        checkout_strategy);
-
 G_END_DECLS
 
 #endif /* __GGIT_SUBMODULE_UPDATE_OPTIONS_H__ */
-- 
cgit v0.12

